CMakeLists.txt

```diff
@@ -45,13 +50,21 @@ endif()
 
 include(SetCompilerFlags)
 
+string(TOUPPER ${CMAKE_BUILD_TYPE} CMAKE_BUILD_TYPE_UPPER)
```

Comment:
Since we're referencing `CMAKE_BUILD_TYPE` here, we should be sure it's set somewhere above. I'd suggest adding the following:
```
if (NOT CMAKE_BUILD_TYPE)
  set(CMAKE_BUILD_TYPE "Release" CACHE STRING "Build type" FORCE)
endif()
```
in case the user or outside project hasn't declared their build type.

Reply:
Good point, will add this.

Reply:
Done.